File: 2019/python/Day01_Part2.py
import math

def ammount_of_fuel(mass):
    # devide by 3
    # round down
    # subtract 2

    # do some recursion stuff here
    m = math.floor(mass/3) - 2
    if m <= 0:
        return 0
    else:
        return m + ammount_of_fuel(m)


def total_fuel(my_list):
    if my_list == []:
        return 0
    else:
        return ammount_of_fuel(my_list[0]) + total_fuel(my_list[1:])


# input data from text file
with open('PuzzleInput_Day01.txt', 'r') as f:
    lines = (line.strip() for line in f if line)
    puzzle_input = [float(line) for line in lines]


print("the total fuel is:", total_fuel(puzzle_input))

File: 2019/c++/src/Day01_Part2.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath>

int fuel_mass(const int& the_mass)
{   
    const int tmp_mass = std::floor( the_mass / 3 ) - 2;
    // base case
    if ( tmp_mass <= 0) { return 0; };
    // recursion
    return tmp_mass + fuel_mass( tmp_mass );
}

int main()
{
    int fuel_requirement = 0;
    int mass;
    std::ifstream PuzzleInput("include/Day_01_Input.txt");

    while (PuzzleInput >> mass)
    {
        fuel_requirement += fuel_mass(mass);
    }

    std::cout << fuel_requirement << std::endl;
}